ioos_catalog/tasks/reindex_services.py

```diff
@@ -89,91 +89,77 @@ def reindex_services(filter_regions=None, filter_service_types=None):
-                                s.updated      = datetime.utcnow()
-                                s.contact      = unicode(contact_email)
-                                s.metadata_url = metadata_url
-
-                                # grab opendap form url if present
-                                if s.service_type == 'DAP':
-                                    possible_refs = [r['url'] for r in record.references if r['scheme'] == opendap_form_schema]
-                                    if len(possible_refs):
-                                        s.extra_url = unicode(possible_refs[0])
-
-                                # if we see the service, this is "Active", unless we've set manual (then we don't touch)
-                                if not s.manual:
-                                    s.active = True
-
-                                s.save()
+                                url = unicode(ref['url'])
+                            elif erddap_match:
+                                test_url = (erddap_match.group(1) +
+                                                '.iso19115')
+                                req = requests.get(test_url)
+                                # if we have a valid ERDDAP metadata endpoint,
+                                # store it.
+                                if req.status_code == 200:
+                                    metadata_url = unicode(test_url)
+                            # next record if not one of the previously mentioned
+                            else:
+                                continue
+                            # end metadata find block
+                            s = db.Service.find_one({'data_provider':
+                                                        unicode(region),
+                                                        'url': url})
+                            if s is None:
+                                s               = db.Service()
+                                s.url           = url
+                                s.data_provider = unicode(region)
+                                s.manual        = False
+                                s.active        = True
+
+                                new_services.append(s)
+                            else:
+                                # will run twice if erddap services have
+                                # both .html and .graph, but resultant
+                                # data should be the same
+                                update_services.append(s)
+
+                            s.service_id   = unicode(name)
+                            s.name         = unicode(record.title)
+                            s.service_type = unicode('DAP' if erddap_match
+                                                        else next((k for k,v in services.items() if v == ref["scheme"])))
+                            s.interval     = 3600 # 1 hour
```

Comment:
Not now but in a follow up, could you make this configurable? once we move everything to configure.yml